.github/workflows/tests.yml

```diff
@@ -494,6 +494,211 @@ jobs:
           name: lxd-clients-${{ runner.os }}
           path: bin/
 
+  ui:
+    name: UI e2e tests
+    runs-on: ubuntu-latest
+    needs: [code-tests, documentation]
+    if: ${{ github.repository == 'canonical/lxd' && github.event_name == 'push' }}
+    env:
+      LXD_DIR: "/var/lib/lxd"
+      LXD_OIDC_CLIENT_ID: ${{ secrets.LXD_UI_OIDC_CLIENT_ID }}
+      LXD_OIDC_ISSUER: ${{ secrets.LXD_UI_OIDC_ISSUER }}
+      LXD_OIDC_AUDIENCE: ${{ secrets.LXD_UI_OIDC_AUDIENCE }}
+      LXD_OIDC_USER: ${{ secrets.LXD_UI_OIDC_TEST_USER }}
+      LXD_OIDC_PASSWORD: ${{ secrets.LXD_UI_OIDC_TEST_PASSWORD }}
+      LXD_OIDC_GROUPS_CLAIM: "lxd-idp-groups"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Tune disk performance
+        uses: ./.github/actions/tune-disk-performance
+
+      - name: Install Go
+        uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed # v5.1.0
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Install dependencies
```

Comment:
I think most if not all of it can be replaced by `.github/actions/install-lxd-builddeps`.

Reply:
good idea :)